### ISSUE TITLE
Data storage approaches

### ISSUE THREAD

**Issue by blelump** [decentralized-identity/keriox#70]
Currently KERIOX uses LMDB under the hood to store the processed events. It is also the main blocker right now of going the WASM way. There are alternatives, like already proposed https://docs.rs/sled/0.34.6/sled/ , but we should decide how to tackle that.

Basically we have two options:

- Change the database driver and use Sled instead of LMDB;
- Continue with support for LMDB, but add additional driver for Sled.

Both are from the same family of embedded databases.

Basically the database abstraction is already [there](https://github.com/decentralized-identity/keriox/blob/master/src/database/mod.rs), so it is a matter of implementing another driver. The question remains whether it makes sense to keep both drivers.

Thoughts?

**Comment by 35359595** [comment]
We've discussed this with [Edyta](@edytapawlak) today and came to next conclusions:

1. LMDB implementation was "cloned" from keripy to prep for some demos;
2. Current implementation state is mixed storage layer with business logic layer, which is bad;
3. Current implementation in many cases have keys bigger than actual values behind them with expensive calculations to get those keys, which is bad;
4. Before storing data to DB in LMDB structs are been teared apart for the parameters and then returned in pieces or blobs, which is not ideal;
5. Only visible reason to have LMDB is to have option to work with the same db file as keripy does, but that implies trusting all the conclusions done by other implementation. Much better way is to replay everything through KEL + KERL + TEL, etc.

Outcome direction forward: As sled abstraction to DB already in place we can easily use it as storage layer and move business logic into processor, where it belongs. Sled will become sole DB provider unless there will be an urge to have similar abstraction layer to it (which is possible), but makes not much sense to support external dependency when we have native DB implementation.
